esmf_regrid/experimental/io.py

```diff
@@ -43,6 +57,49 @@ def _add_mask_to_cube(mask, cube, name):
         cube.add_aux_coord(mask_coord, list(range(cube.ndim)))
 
 
+@contextmanager
+def _managed_var_name(src_cube, tgt_cube):
+    src_coord_names = []
+    src_mesh_coords = []
+    if src_cube.mesh is not None:
+        src_mesh = src_cube.mesh
+        src_mesh_coords = src_mesh.coords()
+        for coord in src_mesh_coords:
+            src_coord_names.append(coord.var_name)
+    tgt_coord_names = []
+    tgt_mesh_coords = []
+    if tgt_cube.mesh is not None:
+        tgt_mesh = tgt_cube.mesh
+        tgt_mesh_coords = tgt_mesh.coords()
+        for coord in tgt_mesh_coords:
+            tgt_coord_names.append(coord.var_name)
+
+    try:
+        for coord in src_mesh_coords:
+            coord.var_name = "_".join([SOURCE_NAME, "mesh", coord.name()])
+        for coord in tgt_mesh_coords:
+            coord.var_name = "_".join([TARGET_NAME, "mesh", coord.name()])
+        yield None
+    finally:
+        for coord, var_name in zip(src_mesh_coords, src_coord_names):
+            coord.var_name = var_name
+        for coord, var_name in zip(tgt_mesh_coords, tgt_coord_names):
+            coord.var_name = var_name
+
+
+def _clean_var_names(cube):
+    cube.var_name = None
+    for coord in cube.coords():
+        coord.var_name = None
+    if cube.mesh is not None:
+        cube.mesh.var_name = None
+        for coord in cube.mesh.coords():
+            coord.var_name = None
+        for con in cube.mesh.connectivities():
+            con.var_name = None
+    return cube
+
+
 def save_regridder(rg, filename):
     """
     Save a regridder scheme instance.
```

Comment:
Outside the changes so I can't comment directly, but the following section is now out of date.
```
 Saves either a
    :class:`~esmf_regrid.experimental.unstructured_scheme.GridToMeshESMFRegridder`
    or a
    :class:`~esmf_regrid.experimental.unstructured_scheme.MeshToGridESMFRegridder`.
    Parameters
    ----------
    rg : :class:`~esmf_regrid.experimental.unstructured_scheme.GridToMeshESMFRegridder` or :class:`~esmf_regrid.experimental.unstructured_scheme.MeshToGridESMFRegridder`
        The regridder instance to save.
```

Needs updating to reference the correct supported regridder types

Reply:
👍